src/components/ActionButton/ToggleActionButton.ts

```diff
@@ -0,0 +1,182 @@
+import { ComponentOptions, IResultsComponentBindings, Component, Initialization } from 'coveo-search-ui';
+import { ActionButton } from './ActionButton';
+
+export interface IToggleActionButtonOptions {
+    activatedIcon: string;
+    activatedTooltip: string;
+    icon: string;
+    tooltip: string;
+    click?: () => void;
+    activate?: () => void;
+    deactivate?: () => void;
+}
+
+export class ToggleActionButton extends Component {
+    static ID = 'ToggleActionButton';
+    static ACTIVATED_CLASS_NAME = 'coveo-toggleactionbutton-activated';
+
+    static options: IToggleActionButtonOptions = {
+        /**
+         * Specifies the button icon when the button is activated.
+         *
+         * Default is the empty string.
+         *
+         * For example, with this SVG markup:
+         *
+         * ```xml
+         * <svg width="1em" height="1em">...</svg>
+         * ```
+         *
+         * The attribute would be set like this:
+         *
+         * ```html
+         * <button class='CoveoToggleActionButton' data-activated-icon='&lt;svg width=&quot;1em&quot; height=&quot;1em&quot;&gt;...&lt;/svg&gt;'></button>
+         * ```
+         */
+        activatedIcon: ComponentOptions.buildStringOption(),
+
+        /**
+         * Specifies the button tooltip when the button is activated.
+         *
+         * Default is the empty string.
+         *
+         * ```html
+         * <button class='CoveoToggleActionButton' data-activated-tooltip='My activated button tooltip'></button>
+         * ```
+         */
+        activatedTooltip: ComponentOptions.buildStringOption(),
+
+        /**
+         * Specifies the button SVG icon.
+         * Note: The SVG markup has to be HTML encoded when set using the HTML attributes.
+         *
+         * Default is the empty string.
+         *
+         * For example, with this SVG markup:
+         *
+         * ```xml
+         * <svg width="1em" height="1em">...</svg>
+         * ```
+         *
+         * The attribute would be set like this:
+         *
+         * ```html
+         * <button class='CoveoToggleActionButton' data-icon='&lt;svg width=&quot;1em&quot; height=&quot;1em&quot;&gt;...&lt;/svg&gt;'></button>
+         * ```
+         */
+        icon: ComponentOptions.buildStringOption(),
```

Comment:
I would use `deactivatedIcon` for the name of the variable, to have the code be semantically coherent with the activated one.

Reply:
Yes, it makes sense.